src/get-started/create-service.md

```diff
@@ -450,71 +450,52 @@ struct CryptocurrencyApi {
 
 ### API for Transactions
 
-To simplify request processing we add a `TransactionRequest` enum
-which joins both types of our transactions.
-We also implement the `Into<Box<Transaction>>` trait for this enum
-to make sure deserialized `TransactionRequest`s fit into the node channel.
+The core processing logic is essentially the same for both types of transactions:
 
-```rust
-#[serde(untagged)]
-#[derive(Clone, Serialize, Deserialize)]
-enum TransactionRequest {
-    CreateWallet(TxCreateWallet),
-    Transfer(TxTransfer),
-}
+1. Convert JSON input into a `Transaction`
+2. Send the transaction to the channel, so that it will be broadcasted over the
+  blockchain network and included into the block.
+3. Synchronously respond with a hash of the transaction
 
-impl Into<Box<Transaction>> for TransactionRequest {
-    fn into(self) -> Box<Transaction> {
-        match self {
-            TransactionRequest::CreateWallet(trans) => Box::new(trans),
-            TransactionRequest::Transfer(trans) => Box::new(trans),
-        }
-    }
-}
+This logic can be encapsulated in a parameterized method in `CryptocurrencyApi`:
 
+```rust
 #[derive(Serialize, Deserialize)]
 struct TransactionResponse {
     tx_hash: Hash,
 }
-```
-
-To join our handler with the HTTP handler of the web-server we need to implement
-the `wire` method. This method takes the reference to a router.
-In the method below we add
-one handler to convert JSON input into a `Transaction`.
-The handler responds with a hash of the transaction. It also
-sends the transaction to the channel, so that it will be broadcasted over the
-blockchain network and included into the block.
-
-```rust
-impl Api for CryptocurrencyApi {
-    fn wire(&self, router: &mut Router) {
-        let self_ = self.clone();
 
-        let tx_handler = move |req: &mut Request| -> IronResult<Response> {
-            match req.get::<bodyparser::Struct<TransactionRequest>>() {
-                Ok(Some(tx)) => {
-                    let tx: Box<Transaction> = tx.into();
-                    let tx_hash = tx.hash();
-                    self_.channel.send(tx).map_err(ApiError::from)?;
-                    let json = TransactionResponse { tx_hash };
-                    self_.ok_response(&serde_json::to_value(&json).unwrap())
-                }
-                Ok(None) => Err(ApiError::IncorrectRequest(
-                    "Empty request body".into()))?,
-                Err(e) => Err(ApiError::IncorrectRequest(Box::new(e)))?,
+impl CryptocurrencyApi {
+    fn post_transaction<T>(&self, req: &mut Request) -> IronResult<Response>
+    where
+        T: Transaction + Clone + for<'de> Deserialize<'de>,
```

Comment:
This line needs import of `use exonum::encoding::serialize::reexport::Deserialize;` above.

Reply:
Or `use serde::Deserialize;` ([as used in code](https://github.com/exonum/cryptocurrency/blob/714ba3cb21ab5b0d9b8ef2072e99147a3e757cec/src/main.rs#L40)), right?